### PR TITLE
Fix/wallet locked issue

### PR DESCRIPTION
**Pull request by svenvoskamp**
# Changes

- fix: Fix wallet locked autoconnect issue

### CODE REVIEW

File: packages/wagmi/src/connectors/EIP6963Connector.ts [hunk 1]
```diff
@@ -74,10 +74,18 @@ export class EIP6963Connector extends InjectedConnector {
       if (!eip6963Wallet || connectedEIP6963Rdns !== eip6963Wallet.info.rdns) {
         return true
       }
+    }
+
+    if (
+      (eip6963Wallet && eip6963Wallet.provider._state?.isUnlocked === undefined) ||
+      (eip6963Wallet && eip6963Wallet.provider._state?.isUnlocked)
```
Comment (tomiir) [inline packages/wagmi/src/connectors/EIP6963Connector.ts:81]: wouldn't it be the same to check 
if (eip6963Wallet && eip6963Wallet.provider?._state?.isUnlocked === true) { return true }

**Review by glitch-txs — COMMENTED** [review]
This is not a standard and some wallets might not have it, let's rather check for eth_account which should return an empty array when locked or disconnected.

File: packages/wagmi/src/connectors/EIP6963Connector.ts [hunk 2]
```diff
@@ -74,10 +74,18 @@ export class EIP6963Connector extends InjectedConnector {
       if (!eip6963Wallet || connectedEIP6963Rdns !== eip6963Wallet.info.rdns) {
```
Comment (glitch-txs) [inline packages/wagmi/src/connectors/EIP6963Connector.ts:74]: shouldn't this be the other way around?

Comment (glitch-txs) [reply packages/wagmi/src/connectors/EIP6963Connector.ts:74]:
```suggestion
      if (eip6963Wallet && connectedEIP6963Rdns === eip6963Wallet.info.rdns) {
        const accounts = (await getProvider()).request({ method: 'eth_accounts' })
        if(accounts.length) return true
      }
```

could you try this?

**Pull request closed without merging**